vmm/src/cpu.rs

```diff
@@ -680,14 +680,23 @@ impl CpuManager {
             .sgx_epc_region()
             .as_ref()
             .map(|sgx_epc_region| sgx_epc_region.epc_sections().values().cloned().collect());
+
+        #[allow(unused_mut)]
```

Comment:
Why do you need to allow this? Can we assign the value within an if condition?

Reply:
When compiled without `mshv` feature, say `kvm` only, the code in the if condition doesn't exist. Thus, Rust would produce a warning about unused mut.

One can of course assign the topology explicitly by removing the `mshv` requirement, i just didn't want to change the existing behavior. For `mshv`, this piece is required.

Either of solutions would work. It seemed to me, that not changing the existing behavior under `kvm` makes sense. But we could go for the first option, too.

Thanks

Reply:
Hmm, maybe i oversee what you mean. Which kind of if construct do you suggest?

Thanks

Reply:
Not sure if this makes sense:

let topology = if let Some(_topology) = self
        .config
        .topology
        .clone()
        .map(|t| (t.threads_per_core, t.cores_per_die, t.dies_per_package)) {
            Some(_topology)
        } else {
            Some((1, self.boot_vcpus(), 1))
        };

Considering for KVM  it always goes to if condition.